src/lib/ProductImporter.ts

```diff
@@ -532,4 +577,77 @@ export class ProductImporter {
         });
         allProvisioningTasks.forEach(prvTask =>{this.provisioningTasks.push(prvTask)});
     }
+
+    private async retrievePbesToUpdate(conn: Connection, pbes:Array<String>, isStandard: boolean){
+        let allPricebookEntriesTarget= isStandard
+         ? await Queries.queryTargetStdPricebookEntry(conn, this.productList) 
+         : await Queries.queryTargetPricebookEntry(conn, this.productList);       
+        let pbesToUpdate = new Array<Object>();
+        
+        for(let pbe of pbes){
+            for(let pbeEntryTarget of allPricebookEntriesTarget){
+                let productMatch = pbe['Product2']['enxCPQ__TECH_External_Id__c'] === pbeEntryTarget['Product2']['enxCPQ__TECH_External_Id__c'];
+                let pricebookMatch = isStandard
+                ? true
+                : pbe['Pricebook2']['enxCPQ__TECH_External_Id__c'] === pbeEntryTarget['Pricebook2']['enxCPQ__TECH_External_Id__c'];
+                let currencyMatch = pbe['CurrencyIsoCode'] === pbeEntryTarget['CurrencyIsoCode'];
+                if(productMatch && pricebookMatch && currencyMatch){
```

Comment:
I'd create separate method for matching - probably at this code level as a developer I shouldn't care what does it mean that we've got a match